cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamHandler.java

```diff
@@ -200,8 +208,11 @@ public void create(HttpRequest request, HttpResponder responder,
       // TODO: For create successful, 201 Created should be returned instead of 200.
       responder.sendStatus(HttpResponseStatus.OK);
     } catch (IllegalArgumentException e) {
-      responder.sendString(HttpResponseStatus.BAD_REQUEST,
-                           e.getMessage());
+      LOG.error("Failed to create stream {}", stream, e);
+      responder.sendString(HttpResponseStatus.BAD_REQUEST, e.getMessage());
+    } catch (NotFoundException e) {
+      LOG.error("Failed to create stream {}", stream, e);
+      responder.sendString(HttpResponseStatus.BAD_REQUEST, e.getMessage());
```

Comment:
what happens to other exceptions here? get propagated as ``INTERNAL_SERVER_ERROR``? Maybe add a catch block for ``Exception`` to do that explicitly?


Come to think of it, is it better to catch these 'specific' exceptions where they are thrown, log, respond and return? Meaning ``IllegalArgumentException`` would move to line 200, ``NotFoundException`` to line 203. Then you would catch a generic ``Exception`` here to send ``INTERNAL_SERVER_ERROR``?